lib/liquid/block_body.rb

```diff
@@ -23,9 +62,20 @@ def parse(tokenizer, parse_context)
           unless token =~ FullToken
             raise_missing_tag_terminator(token, parse_context)
           end
-          tag_name = $1
-          markup = $2
-          # fetch the tag from registered blocks
+          tag_name = $2
+          markup = $4
+
+          if parse_context.line_number
+            # newlines inside the tag should increase the line number,
+            # particularly important for multiline {% liquid %} tags
+            parse_context.line_number += $1.count("\n".freeze) + $3.count("\n".freeze)
+          end
+
+          if tag_name == 'liquid'.freeze
+            liquid_tag_tokenizer = Tokenizer.new(markup, line_number: parse_context.line_number, for_liquid_tag: true)
+            next parse(liquid_tag_tokenizer, parse_context, &block)
+          end
+
```

Comment:
Why is this a special case instead of a new Liquid::Tag?

{% liquid %} requires code in BlockBody#parse anyway. I initially wrote this with a new Liquid::Tag, but found the interaction more difficult to follow with the extra class. This approach also avoids creating the Tag object and an extra BlockBody object for each instance. More importantly, appending to the existing BlockBody avoids _any_ loss of render performance for using {% liquid %}

Reply:
Is this really a special case?  I can think of other places where we could use inline parsing if it were generic:
* `echo` tag to avoid loss of render performance from converting variables to use a `{% liquid %}` tag
* `raw` tag so that it has the same render performance as text outside of a tag
* `comment` tag so that it has no render overhead

Do we care that much about performance of code already optimized by liquid-c?